### ISSUE TITLE
[0.9.0 staging-1605] - 'Currency in Circulation' not fit for purpose

### ISSUE THREAD

**Issue by trickenso** [StrangeLoopGames/EcoIssues#16776]
Currency in Circulation includes all minted currency, does not specify what 'in circulation' means, and includes currency in Treasury, making this statistic 'Total amount of minted currency' and nothing to do with what is 'in circulation'.

This is also a problem for laws like #16772, since they will calculate based on total currency minted, not currency held by players - if the treasury has minted a large buffer to manage the money supply, this calculation is useless.

It is impossible, for example, to ask the question "Is this player in the top 10% in terms of wealth", since you cannot compare one player's accounts with the rest of the player base in aggregate.

Similarly, this means you cannot compare whether Carpenters are doing better than Smelters without custom stats, as you cannot calculate the average wealth of a particular set of players.

Having Currency in Circulation exclude government funds would be a start, but more granularity and aggregation of 'wealth' and other economic statistics would allow some pretty fundamental questions to be asked.

**Comment by Engimage** [comment]
Currency in circulation makes sense.
However we are just lacking the functionality to evaluate currency on player accounts.
And yes, would be nice to have summary of currency owned by a certain demographic like "Active"

**Comment by trickenso** [comment]
> Currency in circulation makes sense.

How so, when there is no way to exclude funds that are not 'in circulation', for example: the Treasury.

If the Federal Reserve prints seventy million dollars and leaves it in a mattress, is it 'in circulation'? According to Eco: yes, it is.

So no, I don't think it 'makes sense'.

**Comment by MishkaFur** [comment]
> 
> 
> Currency in circulation makes sense.
> However we are just lacking the functionality to evaluate currency on player accounts.
> And yes, would be nice to have summary of currency owned by a certain demographic like "Active"

It could be good to perhaps have it  as two things then? as this would allow for more indepth laws doing math off percentages based off how much the players have gotten out of the govt.

Total Currency Minted
Currency in use (ie what the players have)

**Comment by trickenso** [comment]
> Total Currency Minted
> Currency in use (ie what the players have)

This.
Also ideally some way to customise the calculation - "Currency in use by X demographic or title"

**Comment by Deantwo** [comment]
Currency held by demographic. That might create sone issues if any of the players in that demographic has shared bank accounts.

Not an impossible issue to solve, but it shouldn't be forgotten. For example get list of all bank accounts controlled players in the demographic, and then remove duplicates bank accounts before summing the amount of currency in the bank accounts.

The old method of `all currency - treasury currency` worked fine. We just need a way to get the amount in a specific bank account, or even bank accounts owned by "government". That way we can just do the math ourselves.

**Comment by trickenso** [comment]
> players in that demographic has shared bank accounts.

Agree it's a problem to be solved, and also that it's not a massive head-scratcher.

> remove duplicate bank accounts before summing the amount of currency in the bank accounts.

Remove duplicates, and optionally split amount held by the number of account holders (to account for X of Y account holders falling within the filter). How is this done for the current 'Wealth' calculation? Is it just all accessible funds, even if the account is held by multiple people?
And/or an option to exclude shared accounts?

> The old method of `all currency - treasury currency` worked fine. We just need a way to get the amount in a specific bank account, or even bank accounts owned by "government". That way we can just do the math ourselves.

And rename it to something other than 'currency in circulation', if that's not what it is.

> Of course gonna be interesting with the plans for multiple governments in v10.0, since I assume that could mean multiple treasury bank accounts.

Or governments could have the option to have no treasury account at all - or five. The hard-coded single treasury account is far from ideal when the rest of government is trying to be so flexible.

**Comment by WeaselDog** [comment]
What we really need is a filter when checking the Currency in Circulation to exclude government accounts. This will add much more flexibility for laws and other purposes. 

In 9.0, we have added the option to create unlimited government accounts that are exempt from personal wealth calculations, which applies to the treasury by default. It is found in the advanced settings when accessing a bank.

As a side note, if you are running a government and have control of the mint and treasury, you can mint new currency when needed and dump currency at the treasury to remove it from circulation.

**Comment by trickenso** [comment]
> What we really need is a filter when checking the Currency in Circulation to exclude government accounts. This will add much more flexibility for laws and other purposes.

Agreed - a more general filter to select which accounts/types of accounts would solve all kinds of things. A check to exclude 'government' would solve a smaller number of cases, but still helpful.

> In 9.0, we have added the option to create unlimited government accounts that are exempt from personal wealth calculations, which applies to the treasury by default. It is found in the advanced settings when accessing a bank.

Which raises the smaller question of why these aren't automatically excluded from 'in circulation' - or another stat/option added to account for this. If I print twenty billion spondulicks and put them in a mattress in the central bank vault, they are not 'in circulation'.

> As a side note, if you are running a government and have control of the mint and treasury, you can mint new currency when needed and dump currency at the treasury to remove it from circulation.

This doesn't help if you need a reserve of currency to supply subsidies, or if you are not online to immediately dump currency after a tax law runs. I hope manually printing/dumping currency to keep up with every time funds get allocated/taxed is not the intended solution.

**Comment by WeaselDog** [comment]
By default, the treasury is not a central bank. It is where tax money goes into and where government funds are spent from. In modern economics, a central bank is independent from the legislature and it controls how the money supply by issuing bonds and loans to licensed private banks. Congress/votes decide how the tax money is spent.

Using a mint to have a limitless amount of funds in the treasury is one way to run a government, but this can lead to significant inflation and there are benefits to players to be able to track when this is happening. In addition, some people use a backed currency instead of a fiat currency. A government with a balanced budget would have their expenditures covered by taxes, though there is no "right" way to manage a money supply.

**Comment by trickenso** [comment]
What has any of that got to do with being the inability to exclude accounts from currency calculations? I've re-read your reply multiple times, and it seems increasingly likely you're completely missing my point. Let me try and clarify:

The *only* currency values available for laws are "Currency in Circulation" which is really "All Currency Minted" and "Wealth" which is per player and cannot be aggregated.

Some examples of things that are missing and cannot be legislated on: Treasury Amount, Total Government Holdings, Average Player Wealth, Total Player Wealth, Wealth by Demographic, and many other things.

My point: It might be useful to add some of this.

The addition of these things does not limit anyone's ability to do any of the things you've mentioned or say anything about play style in any way. The omission of these things does not "allow all different styles of currency".

Before citing modern economics, you might want to make it possible to sum the value of two accounts.

**Comment by WeaselDog** [comment]
I was just explaining why by default Currency in Circulation counts the Treasury account. We do want to add the filter to exclude government accounts from laws.

Some of these things are already possible, but the rest would be enabled by enabled by having a filter for government accounts, minus the Wealth by Demographic
